packages/opentelemetry-resources/src/platform/node/detectors/AwsEc2Detector.ts

```diff
@@ -0,0 +1,99 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as http from 'http';
+import { Resource } from '../../../Resource';
+import { CLOUD_RESOURCE, HOST_RESOURCE } from '../../../constants';
+import { Detector } from '../../../types';
+
+/**
+ * The AwsEc2Detector can be used to detect if a process is running in AWS EC2
+ * and return a {@link Resource} populated with metadata about the EC2
+ * instance. Returns an empty Resource if detection fails.
+ */
+class AwsEc2Detector implements Detector {
+  readonly AWS_INSTANCE_IDENTITY_DOCUMENT_URI =
+    'http://169.254.169.254/latest/dynamic/instance-identity/document';
```

Comment:
Is this url documented anywhere? Seems like a link to the documentation would be helpful if this ever breaks

Reply:
I'll add a comment pointing to: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html